bot/main.py

```diff
@@ -41,87 +41,94 @@ def main():  #
+                    # times before audit interval hits, so we can ensure we maintain user database accurately)
+                    bot_instance.trigger_user_audit()
+
+                    # Set audit schedule job to run in X days
+                    schedule.every(config.audit_interval).days.do(bot_instance.trigger_user_audit)
+
+                    LOG.info("BOT now online,sending broadcast.")
+                    bot_instance.broadcastMessage()  # Send initial message into channel
+
+                    # debug
+                    # BOT.setResetroster(main_ts3conn, "2020-04-01",
+                    #                    red = ["the name with the looooong name"],
+                    #                    green = ["another really well hung name", "len", "oof. tat really a long one duuuude"],
+                    #                    blue = ["[DUST] dude", "[DUST] anotherone", "[DUST] thecrusty dusty mucky man"],
+                    #                    ebg = [])
+                    # testguilds = [
+                    #     ("Unleash Chaos", "uC", "uC"),
+                    #     ("Requiem of Execution", "RoE", "RoE"),
+                    #     ("Zum Henker", "ZH", "ZH"),
+                    #     ("Formation Wolke", "Zerg", "Zerg."),
+                    #     ("Zergs Rebellion", "Zerg", "Zerg"),
+                    #     ("Flussufer Beach Boys", "FBB", "FBB"),
+                    #     ("Ups Falsche Taste", "UPS", "UPS"),
+                    #     ("Rising River", "Side", "Side"),
+                    #     ("Demons of Dawn", "DoD", "DoD")
+                    # ]
+
+                    # for gname, gtag, ggroup in testguilds:
+                    #    BOT.removeGuild(gname, gtag, ggroup)
+                    #    BOT.createGuild(gname, gtag, ggroup, ["len.1879", "jey.1111"])
+
+                    # Forces script to loop forever while we wait for events to come in, unless connection timed out. Then it should loop a new bot into creation.
+                    LOG.info("BOT now idle, waiting for requests.")
+                    while ts_facade.is_connected():
+                        # auditjob + keepalive check
+                        schedule.run_pending()
+                        response: TS3Event
+                        try:
+                            response = ts_facade.wait_for_event(timeout=config.bot_sleep_idle)
+                            event_type: str = response.event
+                            event_data = response.parsed[0]
+                            if event_type == 'notifytextmessage':
+                                if "msg" in event_data:
+                                    # text message
+                                    bot_instance.messageEventHandler(event_data)  # handle event
+                            elif event_type == 'notifycliententerview':
+                                if event_data["client_type"] == '1':  # server query client
+                                    continue
```

Comment:
Is there a reason for this style, instead of 
```python
if event_data["client_type"] != '1':
    bot_instance.loginEventHandler(event_data)  # handle event
```
?

Reply:
I found it much more readable as it's basically an explicit skipping of this event

Reply:
I strongly oppose any use of `continue` (at all), `break` (outside of `switch`), and `return` (unless as very last statement of a function or as early exit at the start of a function), which are all [`goto`s in disguise](https://homepages.cwi.nl/~storm/teaching/reader/Dijkstra68.pdf). If we were to add any sort of cleanup code at the end of the loop, it would be skipped altogether for `client_type == 1`. I have suggested this before and it wasn't an urgent issue back then either, but I find this style to be a gateway for very hard to find bugs.

Reply:
fixed